### PR TITLE
Proxy Support & Set emoji as favicon

### LINKED ISSUE #54: Add support for proxies.
Attempting to decode a transaction via Etherscan when a proxy is involved results in failing to decode.  We need to figure out how to get Etherscan to return the ABI for the proxy target.

### LINKED ISSUE #65: Set favicon to bouquet emoji

### PR DESCRIPTION
**Pull request by 0xjimmy**
Changes favicon to bouquet emoji, Fixes #65 
Adds proxy support, if proxy implementation is verified on Etherscan, Fixes #54

### CODE REVIEW

**Comment by 0xjimmy** [comment]
BNB uses a proxy, you can use it to test by decoding an approval

File: app/ts/components/Transactions.tsx [hunk 1]
```diff
@@ -58,13 +58,26 @@ export const Transactions = ({
...
+						}.etherscan.io/api?module=contract&action=getsourcecode&address=${getAddress(address.toLowerCase())}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`,
 					),
 				),
 			)
-			const abis = await Promise.all(requests.map((request) => request.json()))
-			interfaces.value = abis.reduce((acc, curr: { status: string; result: string }, index) => {
-				if (curr.status === '1') return { ...acc, [`${uniqueAddresses[index]}`]: new Interface(curr.result) }
+			const contracts = await Promise.all(requests.map((request) => request.json()))
+			const abis: (string | undefined)[] = []
+			for (let contract of contracts) {
```
Comment (KillariDev) [inline app/ts/components/Transactions.tsx:67]:
```suggestion
			for (const contract of contracts) {
```

File: app/ts/components/Transactions.tsx [hunk 2]
```diff
@@ -58,13 +58,26 @@ export const Transactions = ({
...
+				if (!contract || !contract.status || !contract.result || contract.status !== '1') abis.push(undefined)
+				else {
+					if (contract.result[0].Proxy === '0') abis.push(contract.result[0].ABI)
+					else {
+						const implReq = await fetch(`https://api${appSettings.peek().relayEndpoint === MEV_RELAY_GOERLI ? '-goerli' : ''}.etherscan.io/api?module=contract&action=getabi&address=${getAddress(contract.result[0].Implementation.toLowerCase())}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`)
+						const implResult = await implReq.json() as { status: '1' | '0', result?: string }
+						abis.push(implResult.status === "1" && implResult.result ? implResult.result : undefined)
```
Comment (KillariDev) [inline app/ts/components/Transactions.tsx:74]:
```suggestion
						abis.push(implResult.status === '1' && implResult.result ? implResult.result : undefined)
```

File: app/ts/components/Transactions.tsx [hunk 3]
```diff
@@ -58,13 +58,26 @@ export const Transactions = ({
...
 						`https://api${appSettings.peek().relayEndpoint === MEV_RELAY_GOERLI ? '-goerli' : ''
-						}.etherscan.io/api?module=contract&action=getabi&address=${getAddress(address.toLowerCase())}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`,
+						}.etherscan.io/api?module=contract&action=getsourcecode&address=${getAddress(address.toLowerCase())}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`,
 					),
 				),
 			)
-			const abis = await Promise.all(requests.map((request) => request.json()))
-			interfaces.value = abis.reduce((acc, curr: { status: string; result: string }, index) => {
-				if (curr.status === '1') return { ...acc, [`${uniqueAddresses[index]}`]: new Interface(curr.result) }
+			const contracts = await Promise.all(requests.map((request) => request.json()))
```
Comment (KillariDev) [inline app/ts/components/Transactions.tsx:65]: we should be parsing the request that it matches expected type

File: app/ts/components/Transactions.tsx [hunk 4]
```diff
@@ -58,13 +58,26 @@ export const Transactions = ({
...
+			const abis: (string | undefined)[] = []
+			for (let contract of contracts) {
+				if (!contract || !contract.status || !contract.result || contract.status !== '1') abis.push(undefined)
+				else {
+					if (contract.result[0].Proxy === '0') abis.push(contract.result[0].ABI)
+					else {
+						const implReq = await fetch(`https://api${appSettings.peek().relayEndpoint === MEV_RELAY_GOERLI ? '-goerli' : ''}.etherscan.io/api?module=contract&action=getabi&address=${getAddress(contract.result[0].Implementation.toLowerCase())}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`)
+						const implResult = await implReq.json() as { status: '1' | '0', result?: string }
```
Comment (KillariDev) [inline app/ts/components/Transactions.tsx:73]: we should be parsing this with funtypes and not assuming its correct

**Review by KillariDev — CHANGES_REQUESTED** [review]
we should be parsing the results we are getting from various API's to check that they return what we expect

**Review by KillariDev — APPROVED** [review]

**Pull request merged**